data/caasp/update.sh

```diff
@@ -123,12 +143,56 @@ elif [ ! -z "${INSTALL:-}" ]; then
         fi
     done < qam_packages.txt
 
+    if [ ${reboot_required} -eq 1 ]; then
+        # Disable the update repo, so to check if this package is old or new
+        $runner "zypper mr -d UPDATE"
+        new_package=0
+        while read pkg; do
+            if ! zypper se $pkg; then
+                echo "$pkg " >> new_package.txt
+                new_package=1
+            fi
+        done < not_installed.txt
+
+        if [ ${new_package} -eq 1 ]; then
+
+            # Enable the update repo to install the new package
+            $runner "zypper mr -e UPDATE"
+            $runner "zypper lr UPDATE | grep Enabled | grep Yes"
+            echo "The following new packages are going to be installed: $(cat new_package.txt)"
+
+            # Install the packages
+            $runner "/usr/sbin/transactional-update salt pkg install -y $(cat new_package.txt)"
+
+            # Refresh the Grains
+            $srun '*' saltutil.refresh_grains
+
+            # Orchestrate the reboot via Velum to complete the installation
+            rm not_installed.txt
+            exit 100
+        fi
+    fi
+
+elif [ ! -z "${INSTALL:-}" ]; then
+    # In case there was a new package, it must be installed by now. Clear the state.
+    if [ -f not_installed.txt ]; then
+         rm not_installed.txt
+    fi
+    # Check if there are any packages included into the QAM incident that are not pre-installed
+    reboot_required=0
```

Comment:
That's the point. Every time I parse the packages, I expect different status depending on the maintenance update: there might be packages needed to be installed first -- this means that the *second time* I parse them, these packages should be installed, so they should *not* end-up in `not_installed.txt`. The third time it depends of there's a new package or not. Of course in case all packages are pre-installed and there are no new packages, I am parsing 2 additional times for no reason, so yes I see your point. Maybe there's another way. I am open to suggestions.